controller/space_codebases_test.go

```diff
@@ -160,6 +159,31 @@ func (rest *TestSpaceCodebaseREST) TestFailCreateCodebaseNotAuthorized() {
 	test.CreateSpaceCodebasesUnauthorized(t, svc.Context, svc, ctrl, uuid.NewV4(), ci)
 }
 
+func (rest *TestSpaceCodebaseREST) TestFailCreateCodebaseExistingCodebase() {
+	t := rest.T()
+	resource.Require(t, resource.Database)
+
+	s := rest.createSpace(testsupport.TestIdentity.ID)
+	spaceId := s.ID
+	svc, ctrl := rest.SecuredController()
+
+	repo := "https://github.com/fabric8-services/fabric8-wit.git"
+	stackId := "stackId"
+
+	// add first codebase
+	spaceCodebaseContext := createSpaceCodebase(repo, &stackId)
```

Comment:
or use test_fixture here as well. it will take care of all creation logic. and use the same URL from that created object to create next object.

Reply:
Please excuse me but I don't understand by `test_fixture`, what are you referring to?

Reply:
What I meant was, you can tell fixture to create one codebase for this test and no need to call create API (first time) inside the test.
Once you have the codebase created in DB, just try to create another codebase with same data using create API this time (which this test should do)
Like you have created codebase here with test_fixture, https://github.com/fabric8-services/fabric8-wit/pull/1925/files#diff-bcdebbbc92dbb03ef502a37a78d2bc87R172
you test code will look like 
```
fxt := tf.NewTestFixture(test.T(), test.DB, tf.Codebases(1))

// add another codebase with same url, should fail
 spaceCodebaseContext = createSpaceCodebase(fxt.Codebase[0].repo, fxt.Codebase[0].stackId)
 _, err := test.CreateSpaceCodebasesConflict(t, svc.Context, svc, ctrl, fxt.Codebase[0].spaceId, spaceCodebaseContext)
 require.NotNil(t, err)

Reply:
done.

thanks for helping on this offline :-)